packages/editor/src/components/browser/categorie/CatPathChooser.tsx

```diff
@@ -1,21 +1,157 @@
-import { useState } from 'react';
-import { Input } from '../../widgets';
+import { useEffect, useMemo, useState } from 'react';
+import { ExpandableCell, SelectRow, Table, TableCell } from '../../widgets';
 import type { UseBrowserImplReturnValue } from '../useBrowser';
 import { IvyIcons } from '@axonivy/editor-icons';
 import type { BrowserValue } from '../Browser';
+import { useEditorContext, useMeta } from '../../../context';
+import {
+  useReactTable,
+  type ColumnDef,
+  type RowSelectionState,
+  getCoreRowModel,
+  getFilteredRowModel,
+  flexRender
+} from '@tanstack/react-table';
+import type { CategoryPathMeta, Type1 } from '@axonivy/inscription-protocol';
 export const CAT_PATH_CHOOSER_BROWSER_ID = 'catPath' as const;
 
-export const useCatPathChooserBrowser = (): UseBrowserImplReturnValue => {
-  const [value, setValue] = useState<BrowserValue>({ cursorValue: 'category path chooser' });
+export const useCatPathChooserBrowser = (onDoubleClick: () => void, location: string): UseBrowserImplReturnValue => {
+  const [value, setValue] = useState<BrowserValue>({ cursorValue: '' });
+
   return {
     id: CAT_PATH_CHOOSER_BROWSER_ID,
     name: 'Category Path Chooser',
-    content: <CatPathChooserBrowser value={value.cursorValue} onChange={(change: string) => setValue({ cursorValue: change })} />,
+    content: <CatPathChooserBrowser value={value.cursorValue} onChange={setValue} location={location} onDoubleClick={onDoubleClick} />,
     accept: () => value,
     icon: IvyIcons.Label
   };
 };
 
-const CatPathChooserBrowser = (props: { value: string; onChange: (value: string) => void }) => {
-  return <Input {...props} />;
+const CatPathChooserBrowser = (props: {
+  value: string;
+  onChange: (value: BrowserValue) => void;
+  location: string;
+  onDoubleClick: () => void;
+}) => {
+  const { context } = useEditorContext();
+
+  const type = (): Type1 => {
+    const location = props.location.toLowerCase();
+    switch (true) {
+      case location.includes('request') || location.includes('start'):
+        return 'START';
+      case location.includes('case'):
+        return 'CASE';
+      case location.includes('task'):
+        return 'TASK';
+      default:
+        throw new Error('Unknown type');
```

Comment:
are you sure about this?
If this happens maybe the whole UI is broken. Is this what we want?
Or did you try it?
I think we can simply fire a request with an unknown location and the backend should then simply return nothing or something like this...